### PR TITLE
Insert Flag

### LINKED ISSUE #650: Remove unused dependencies
Add this to all pallets and runtime on node side
#![deny(unused_crate_dependencies)]

### PR DESCRIPTION
**Pull request by nakul1010**
Fixes #650.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -66,7 +65,7 @@ pallet-scheduler = { git = "https://github.com/paritytech/substrate", branch = "
 pallet-im-online = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.28", default-features = false }
 pallet-contracts = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.28", default-features = false }
 pallet-timestamp = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.28", default-features = false }
-pallet-bags-list = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.28", default-features = false }
+#pallet-bags-list = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.28", default-features = false }
```
Comment (serhii-temchenko) [inline Cargo.toml]: Please remove if not used

File: pallets/chainbridge/src/tests.rs [hunk 2]
```diff
@@ -11,6 +11,7 @@ use super::{
 	pallet::Event as PalletEvent,
 	*,
 };
+use frame_benchmarking as _;
```
Comment (serhii-temchenko) [inline pallets/chainbridge/src/tests.rs]: Why is it imported like this? 
If you need somehow import benchmarks then I assume you need to do something like [this](https://github.com/Polkadex-Substrate/Polkadex/blob/4e89260855633dccc78026a2b0644951cdd23549/pallets/asset-handler/src/lib.rs#L25-L26) and not here.

File: pallets/ocex/Cargo.toml [hunk 3]
```diff
@@ -35,26 +34,23 @@ pallet-assets = { workspace = true, features = ["std"] }
...
     "frame-system/std",
     "sp-runtime/std",
     "sp-std/std",
-    "sp-core/std",
     "polkadex-primitives/std",
     "ias-verify/std",
     "rust_decimal/std",
     "pallet-timestamp/std",
     "liquidity/std",
+
```
Comment (serhii-temchenko) [inline pallets/ocex/Cargo.toml]: Why?

File: pallets/ocex/Cargo.toml [hunk 4]
```diff
@@ -63,5 +59,6 @@ runtime-benchmarks = [
     "frame-system/runtime-benchmarks",
     "test-utils",
     "liquidity/runtime-benchmarks",
+
```
Comment (serhii-temchenko) [inline pallets/ocex/Cargo.toml]: ?

File: pallets/thea-staking/src/tests.rs [hunk 5]
```diff
@@ -3,6 +3,7 @@ use crate::{
 	session::{Exposure, IndividualExposure, StakingLimits, UnlockChunk},
 	ActiveNetworks, Candidates, CurrentIndex, Error, Hooks, Perbill, Stakers, Stakinglimits,
 };
+use frame_benchmarking as _;
```
Comment (serhii-temchenko) [inline pallets/thea-staking/src/tests.rs]: Same

File: pallets/thea/src/mock.rs [hunk 6]
```diff
@@ -34,7 +34,8 @@ use sp_runtime::traits::AccountIdConversion;
 type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
 type Block = frame_system::mocking::MockBlock<Test>;
 type Balance = u128;
-
+use chainbridge as _;
+use frame_benchmarking as _;
```
Comment (serhii-temchenko) [inline pallets/thea/src/mock.rs]: Same

File: runtime/Cargo.toml [hunk 7]
```diff
@@ -192,15 +182,15 @@ std = [
...
-    "frame-try-runtime/std",
     "pallet-asset-handler-runtime-api/std",
     "thea-staking/std",
     "pallet-nomination-pools/std",
     "pallet-asset-handler-runtime-api/std",
     "thea/std",
     "thea-cross-chain-governance/std",
     "liquidity/std",
+    "pallet-rewards/std"
+
```
Comment (serhii-temchenko) [inline runtime/Cargo.toml]: Same

File: runtime/src/lib.rs [hunk 8]
```diff
@@ -20,6 +20,9 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 // `construct_runtime!` does a lot of recursion and requires us to increase the limit to 256.
 #![recursion_limit = "256"]
+#![deny(unused_crate_dependencies)]
+use pallet_society as _;
+use sp_finality_grandpa as _;
```
Comment (serhii-temchenko) [inline runtime/src/lib.rs]: ? Why is it here?

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]
Missed `liquidity` and `reards` pallets. Any reason to not adding `unused_crate_dependencies` lint there?

File: pallets/ocex/Cargo.toml (hunk 3 above)
Comment (nakul1010) [reply pallets/ocex/Cargo.toml]: removed extra line

File: pallets/ocex/Cargo.toml (hunk 4 above)
Comment (nakul1010) [reply pallets/ocex/Cargo.toml]: removed extra line

File: pallets/thea-staking/src/tests.rs (hunk 5 above)
Comment (nakul1010) [reply pallets/thea-staking/src/tests.rs]: removed benchmarking as mock and test env still not there

File: runtime/Cargo.toml (hunk 7 above)
Comment (nakul1010) [reply runtime/Cargo.toml]: removed extra line

File: pallets/thea/src/mock.rs (hunk 6 above)
Comment (nakul1010) [reply pallets/thea/src/mock.rs]: removed

File: runtime/src/lib.rs (hunk 8 above)
Comment (nakul1010) [reply runtime/src/lib.rs]: removed

**Comment by nakul1010** [comment]
> Missed `liquidity` and `reards` pallets. Any reason to not adding `unused_crate_dependencies` lint there?

The PR was created before both pallets were merged, so it went unnoticed. However, I have added the lint in the latest commit for both pallets.

**Review by serhii-temchenko — APPROVED** [review]

**Review by felixfaisal — APPROVED** [review]
LGTM 🥇

**Pull request merged**